File: 219-ContainDup.py
def containsNearbyDuplicate(nums, k): 
    result = {}
    i = 0
    while i < len(nums): 
        if (nums[i] in result and i - result[nums[i]] <= k):
            return True
        result[nums[i]] = i 
        i += 1

    return False 
            

File: 219-ContainDup.js
var containsNearbyDuplicate = function (nums, k) {
    let obj = {};
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] in obj && i - obj[nums[i]] <= k) {
            return true;   
        } else {
            obj[nums[i]] = i;
        }
    }
    return false;
}

console.log(containsNearbyDuplicate(nums = [1,0,1,1], k = 1))
